sklearn/decomposition/sparse_pca.py

```diff
@@ -283,6 +334,15 @@ def fit(self, X, y=None):
         """
         random_state = check_random_state(self.random_state)
         X = check_array(X)
+
+        if self.normalize_components:
+            self.mean_ = np.mean(X, axis=0)
+            X = X - self.mean_
+        else:
+            warnings.warn("normalize_components should be set to True. "
```

Comment:
I would be more explicit: "normalize_components=False is a backward-compatible setting that implements a non-standard definition of sparse PCA. This compatibility mode will be removed in 0.22."